zyte_spider_templates/spiders/serp.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+from pydantic import BaseModel, Field, field_validator
+from scrapy import Request
+from scrapy.settings import SETTINGS_PRIORITIES, BaseSettings
+from scrapy_spider_metadata import Args
+from w3lib.url import add_or_replace_parameter
+from zyte_common_items import Serp
+
+from ..params import MaxRequestsParam
+from ._google_domains import GoogleDomain
+from .base import BaseSpider
+
+
+class SearchQueriesParam(BaseModel):
+    search_queries: Optional[List[str]] = Field(
+        title="Search Queries",
+        description="Input 1 search query per line (e.g. foo bar).",
+        json_schema_extra={
+            "widget": "textarea",
+        },
+    )
+
+    @field_validator("search_queries", mode="before")
+    @classmethod
+    def validate_search_queries(cls, value: Union[List[str], str]) -> List[str]:
+        """Validate a list of search queries.
+        If a string is received as input, it is split into multiple strings
+        on new lines.
+        """
+        if isinstance(value, str):
+            value = value.split("\n")
```

Comment:
I think we should put this at the bottom after the `if not value` check since having a input value of `""` would result in `[""]` which fails the non-empty check below. We'd want the `ValueError` that says input is empty.

Reply:
Great catch. But in order to also cause `\n` to become `[]`, I will instead move the exception later, to check the contents of `result` instead.